scripts/fiximports/main.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"golang.org/x/tools/imports"
+)
+
+var (
+	// groupByPrefixes is the list of import prefixes that should _each_ be grouped separately.
+	// See: imports.LocalPrefix.
+	groupByPrefixes = []string{
+		"github.com/filecoin-project",
+		"github.com/filecoin-project/lotus",
+	}
+	newline                  = []byte("\n")
+	importBlockRegex         = regexp.MustCompile(`(?s)import\s*\((.*?)\)`)
+	consecutiveNewlinesRegex = regexp.MustCompile(`\n\s*\n`)
+)
+
+func main() {
+	if err := filepath.Walk(".", func(path string, info fs.FileInfo, err error) error {
+		switch {
+		case err != nil:
+			return err
+		case // Skip ffi and test vectors in "./extern" directory and their contents.
+			strings.HasSuffix(path, "extern/filecoin-ffi"),
+			strings.HasSuffix(path, "extern/test-vectors"):
```

Comment:
I'd be a little more strict here. I.e., path is exactly `./extern/...` (in case we have other extern directories eventually). Not critical, but...

Reply:
Hm. Actually, we should probably exclude `extern` entirely. But my point was more: exclude `./extern/` _only_, not everything that happens to end with `extern/...`.